src/core/core.interaction.js

```diff
@@ -39,77 +39,118 @@ function parseVisibleItems(chart, handler) {
 	}
 }
 
+/**
+ * Helper function to check the items at the hovered index on the index scale
+ * @param {Chart} chart - the chart
+ * @param {function} handler - the callback to execute for each visible item
+ * @return whether all scales were of a suitable type
+ */
+function evaluateItemsAtIndex(chart, axis, position, handler) {
+	const metasets = chart._getSortedVisibleDatasetMetas();
+	for (let i = 0, ilen = metasets.length; i < ilen; ++i) {
+		const metaset = metasets[i];
+		const iScale = metaset.controller._cachedMeta.iScale;
+		// TODO: make this more generic. if time and timeseries scale are split then we could check
+		// for existence of getIndexForPixel instead of distribution option. then add getIndexForPixel
+		// to other scales
+		if (!iScale || axis !== iScale.axis || iScale.options.distribution !== 'series') {
+			return false;
+		}
+		const iPixel = iScale.isHorizontal() ? position.x : position.y;
+		const index = iScale.getIndexForPixel(iPixel);
+		const element = metaset.data[index];
```

Comment:
I don't think it's actually possible to return `null` here because this code doesn't get called when we're outside the chart area or else I'd be seeing errors in the console when moving my mouse around the chart

Reply:
`iScale.options.distribution !== 'series'` check can be replaced with a `index === null` check now, to separate concerns.